### ISSUE TITLE
Generating report fails when running from a path with spaces in it

### ISSUE THREAD

**Issue by jcharr1** [TitouanVanBelle/XCTestHTMLReport#150]
Running xchtmlreport 2.0.0 in a directory that has spaces in it (both standalone and through the fastlane plugin) causes it to fail. For example:

```
11:43:21 PM >/Volumes/DATA/My Documents/ILL/iPhone/Projects/ILL for iOS/git/IL101 for iOS/[REDACTED] > 

$ xchtmlreport -r fastlane/test_output/Innovative.test_result 

XCTestHTMLReport 2.0.0
error: unexpected argument Documents/ILL/iPhone/Projects/ILL; use --help to list available arguments
Warning: Can't find invocation record for : fastlane/test_output/Innovative.test_result

Report successfully created at fastlane/test_output/index.html
```

If I copy the test_output folder to a path without spaces in it and try again, it works fine.

**Comment by jordan-romeroporter-earnin** [comment]
This problem also seems to extend to project names with spaces in them. For example:
```
/test_output/Example Project.test_result
```

seems to fail as well

**Comment by jordan-romeroporter-earnin** [comment]
Because my project name has spaces in it, I get this issue:

```
XCTestHTMLReport 2.0.0
Warning: Can't find test reference for action Build "Example ProjectName"

Report successfully created at ./build/DerivedData/test_output/index.html
```

**Comment by jordan-romeroporter-earnin** [comment]
Here's where I believe the code issue is - I'm not sure if replacing empty spaces with a backslash will work, but I'll fork and create a PR at some point today:

https://github.com/TitouanVanBelle/XCTestHTMLReport/blob/develop/Sources/XCTestHTMLReport/Classes/Models/Command.swift

**Comment by jordan-romeroporter-earnin** [comment]
@jcharr1 Have you tried putting quotes around the path?

**Comment by jcharr1** [comment]
@jordan-romeroporter-earnin yes, I tried quotes, but it didn't help. Oddly, I get different errors if I pass the relative path vs the full path.

Relative (same error when not using quotes):
```
xchtmlreport -r 'fastlane/test_output/Innovative.test_result'
XCTestHTMLReport 2.0.0
error: unexpected argument Documents/ILL/iPhone/Projects/ILL; use --help to list available arguments
Warning: Can't find invocation record for : fastlane/test_output/Innovative.test_result

Report successfully created at fastlane/test_output/index.html
```

Full (terminal has full disk access)
```
xchtmlreport -r "/Volumes/DATA/My Documents/ILL/iPhone/Projects/ILL for iOS/git/IL101 for iOS/[REDACTED]/fastlane/test_output/Innovative.test_result"
XCTestHTMLReport 2.0.0
error: unexpected argument Documents/ILL/iPhone/Projects/ILL; use --help to list available arguments
Warning: Can't find invocation record for : /Volumes/DATA/My Documents/ILL/iPhone/Projects/ILL for iOS/git/IL101 for iOS/[REDACTED]/fastlane/test_output/Innovative.test_result
Error: An error has occured while creating the report. Error: Error Domain=NSCocoaErrorDomain Code=512 "The file “Innovative.test_result” couldn’t be saved in the folder “test_output”." UserInfo={NSFilePath=/Volumes/DATA/My Documents/ILL/iPhone/Projects/ILL for iOS/git/IL101 for iOS/[REDACTED]/fastlane/test_output/Innovative.test_result, NSUnderlyingError=0x7ff869c0b650 {Error Domain=NSPOSIXErrorDomain Code=21 "Is a directory"}}
```

**Comment by jordan-romeroporter-earnin** [comment]
Yea you're right @jcharr1 , I wrote that before I even had the results lol. Last time I do that. I'll see if I can figure out a work around

**Comment by jordan-romeroporter-earnin** [comment]
There are two issuesL:

1. How CommandLine parses the command arguments (using space as the delimiter
2. How ResultFile goes about its init:
```
    init(url: URL) {
        self.url = url
        self.relativeUrl = URL(fileURLWithPath: url.lastPathComponent)
        self.file = XCResultFile(url: url)
    }
```
It takes the "lastpathComponent" which ultimately becomes a incorrect string, due to the space. Any ideas on how to fix this?

**Comment by jordan-romeroporter-earnin** [comment]
You would either need to pass in an encoded URL, then decode it, or ...some variation. I don't think anything in swift will do this out of the box :/

**Comment by clindsay** [comment]
@jordan-romeroporter-earnin did you have any luck figuring out how to fix this?

**Comment by tedgonzalez** [comment]
+1

**Comment by chrisballinger** [comment]
Resolved by #165
